src/v/storage/disk_log_impl.cc

```diff
@@ -1529,7 +1529,26 @@ ss::future<> disk_log_impl::truncate_prefix(truncate_prefix_config cfg) {
     vassert(!_closed, "truncate_prefix() on closed log - {}", *this);
     return _failure_probes.truncate_prefix().then([this, cfg]() mutable {
         // dispatch the actual truncation
-        return do_truncate_prefix(cfg);
+        return do_truncate_prefix(cfg)
+          .then([this] {
+              /*
+               * after truncation do a quick refresh of cached variables that
+               * are computed during disk usage calculation. this is useful for
+               * providing more timely updates of reclaimable space through the
+               * health report.
+               */
+              model::timestamp collection_threshold;
+              if (!_manager.config().delete_retention()) {
+                  collection_threshold = model::timestamp(0);
+              } else {
+                  collection_threshold = model::timestamp(
+                    model::timestamp::now().value()
+                    - _manager.config().delete_retention()->count());
+              }
+              return disk_usage_and_reclaimable_space(gc_config(
+                collection_threshold, _manager.config().retention_bytes()));
```

Comment:
hmm, elsewhere `gc_config` is initialized a bit differently (using values from `_config`), is this intended? Maybe this can be encapsulated in a method a-la `get_gc_config()`?

Reply:
The initializations that use `_config` are in `log_manager`, which seems equivalent to `_manager.config()`? Though yeah, this pattern is used a few places -- probably worth unifying.

Reply:
fixed this by factoring out a helper.